src/main/java/com/mongodb/hibernate/jdbc/MongoStatement.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.MongoClient;
+import com.mongodb.hibernate.internal.NotYetImplementedException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.SQLSyntaxErrorException;
+import java.sql.SQLWarning;
+import org.bson.BsonDocument;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * MongoDB Dialect's JDBC {@link java.sql.Statement} implementation class.
+ *
+ * <p>It only focuses on API methods Hibernate ever used. All the unused methods are implemented by throwing exceptions
+ * in its parent class.
+ */
+final class MongoStatement extends StatementAdapter {
+
+    // temporary hard-coded database prior to the db config tech design finalizing
+    private static final String DATABASE = "mongo-hibernate-test";
+
+    private final MongoClient mongoClient;
+    private final MongoConnection mongoConnection;
+    private final ClientSession clientSession;
+
+    private boolean closed;
+
+    public MongoStatement(MongoClient mongoClient, ClientSession clientSession, MongoConnection mongoConnection) {
+        this.mongoClient = mongoClient;
+        this.mongoConnection = mongoConnection;
+        this.clientSession = clientSession;
+    }
+
+    @Override
+    public ResultSet executeQuery(String mql) throws SQLException {
+        checkClosed();
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-21");
+    }
+
+    @Override
+    public int executeUpdate(String mql) throws SQLException {
+        checkClosed();
+        var command = parse(mql);
+        try {
+            mongoConnection.startTransactionIfNeeded();
+            return mongoClient
+                    .getDatabase(DATABASE)
+                    .runCommand(clientSession, command)
```

Comment:
As explained in the PR description, bulk write or JDBC batch processing is in the scope of https://jira.mongodb.org/browse/HIBERNATE-35.